gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/component/GbHeadersToolbar.java

```diff
@@ -0,0 +1,115 @@
+package org.sakaiproject.gradebookng.tool.component;
+
+import org.apache.wicket.AttributeModifier;
+import org.apache.wicket.extensions.markup.html.repeater.data.sort.ISortStateLocator;
+import org.apache.wicket.extensions.markup.html.repeater.data.table.DataTable;
+import org.apache.wicket.extensions.markup.html.repeater.data.table.HeadersToolbar;
+import org.apache.wicket.markup.html.WebMarkupContainer;
+import org.apache.wicket.markup.html.basic.Label;
+import org.apache.wicket.markup.html.list.ListItem;
+import org.apache.wicket.markup.html.list.ListView;
+import org.apache.wicket.spring.injection.annot.SpringBean;
+import org.sakaiproject.gradebookng.business.GbCategoryType;
+import org.sakaiproject.gradebookng.business.GradebookNgBusinessService;
+import org.sakaiproject.gradebookng.tool.model.GradebookUiSettings;
+import org.sakaiproject.gradebookng.tool.pages.GradebookPage;
+import org.sakaiproject.service.gradebook.shared.Assignment;
+import org.sakaiproject.service.gradebook.shared.CategoryDefinition;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class GbHeadersToolbar extends HeadersToolbar {
+
+	@SpringBean(name = "org.sakaiproject.gradebookng.business.GradebookNgBusinessService")
+	private GradebookNgBusinessService businessService;
+
+
+	public GbHeadersToolbar(final DataTable table, final ISortStateLocator stateLocator) {
+		super(table, stateLocator);
+	}
+
+	@Override
+	public void onInitialize() {
+		super.onInitialize();
+
+		final Map<String, Object> gradebookData = ((GbDataTable)getTable()).getGradebookData();
```

Comment:
And here is where I use the `gradebookData` as the toolbar knows its DataTable I can grab them.  This component also knows its page... so maybe?